File: src/main/java/io/dustin/finstage/user/infra/repository/UserRepository.java
package io.dustin.finstage.user.infra.repository;

import io.dustin.finstage.user.infra.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserEntity, Long> {
    Optional<UserEntity> findByEmail(String email);
}

File: src/main/kotlin/io/dustin/finstage/user/infra/repository/UserRepository.kt
package io.dustin.finstage.user.infra.repository

import io.dustin.finstage.user.infra.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findByEmail(email: String): Optional<UserEntity>
}
